tests/gflownet/utils/test_batch.py

```diff
@@ -1325,3 +1330,167 @@ def test__make_indices_consecutive__multiplied_indices_become_consecutive(
+    "env, proxy",
+    [("grid2d", "corners"), ("tetris6x4", "tetris_score_norm")],
+)
+def test__get_rewards_multiple_env_returns_expected_non_zero_non_terminal(
+    env, proxy, batch, request
+):
+    batch_size = BATCH_SIZE
+    env_ref = request.getfixturevalue(env)
+    proxy = request.getfixturevalue(proxy)
+    env_ref = env_ref.reset()
+    env_ref.proxy = proxy
+    env_ref.setup_proxy()
+    env_ref.reward_func = "boltzmann"
+
+    batch.set_env(env_ref)
+
+    # Make list of envs
+    envs = []
+    for idx in range(batch_size):
+        env_aux = env_ref.copy().reset(idx)
+        envs.append(env_aux)
+
+    rewards = []
+    proxy_values = []
+
+    # Iterate until envs is empty
+    while envs:
+        actions_iter = []
+        valids_iter = []
+        # Make step env by env (different to GFN Agent) to have full control
+        for env in envs:
+            parent = copy(env.state)
+            # Sample random action
+            state, action, valid = env.step_random()
+            if valid:
+                # Add to iter lists
+                actions_iter.append(action)
+                valids_iter.append(valid)
+                rewards.append(env.reward(do_non_terminating=True))
+                proxy_values.append(env.proxy(env.state2proxy(env.state))[0])
+        # Add all envs, actions and valids to batch
+        batch.add_to_batch(envs, actions_iter, valids_iter)
+        # Remove done envs
+        envs = [env for env in envs if not env.done]
+
+    rewards_batch = batch.get_rewards(do_non_terminating=True)
+    rewards = torch.stack(rewards)
+    assert torch.equal(
+        rewards_batch,
+        tfloat(rewards, device=batch.device, float_type=batch.float),
+    ), (rewards, rewards_batch)
+    assert ~torch.any(
+        torch.isclose(rewards_batch, torch.zeros_like(rewards_batch))
+    ), rewards_batch
+
+
+@pytest.mark.repeat(N_REPETITIONS)
+# @pytest.mark.skip(reason="skip while developping other tests")
+@pytest.mark.parametrize(
+    "env, proxy",
+    [
+        ("grid2d", "corners"),
+        ("tetris6x4", "tetris_score_norm"),
+        ("ctorus2d5l", "corners"),
+    ],
+)
+def test__get_rewards_parents_multiple_env_returns_expected_non_terminal(
+    env, proxy, batch, request
+):
+    batch_size = BATCH_SIZE
+    env_ref = request.getfixturevalue(env)
+    proxy = request.getfixturevalue(proxy)
+    env_ref = env_ref.reset()
+    env_ref.proxy = proxy
+    env_ref.setup_proxy()
+
+    batch.set_env(env_ref)
+
+    # Make list of envs
+    envs = []
+    for idx in range(batch_size):
+        env_aux = env_ref.copy().reset(idx)
+        envs.append(env_aux)
+
+    rewards_parents = []
+    rewards = []
+
+    # Iterate until envs is empty
+    while envs:
+        actions_iter = []
+        valids_iter = []
+        # Make step env by env (different to GFN Agent) to have full control
+        for env in envs:
+            parent = copy(env.state)
+            done_parent = env.done
```

Comment:
Will a parent ever be done? :thinking:

Reply:
hmm, should never be :) but I think it is more consistent to take this value from the env than hardcode it as False, no?

Reply:
assert?

Reply:
done in 0007bcc4314a3c6980acbb5cacc2af28be753142